src/tiled/command.cpp

```diff
@@ -50,6 +51,34 @@ QString Command::finalCommand() const
     return finalCommand;
 }
 
+QString Command::finalWorkingDir() const
+{
+    QString finalWorkingDir;
+
+    MapDocument *mapDoc = DocumentManager::instance()->currentDocument();
+    if (mapDoc) {
+        const QString mapPath = mapDoc->path();
+        finalWorkingDir.replace(QLatin1String("%executablepath"),
+                                QLatin1String("\"%1\"")).arg(executablePath());
+        finalWorkingDir.replace(QLatin1String("%mappath"),
+                                QLatin1String("\"%1\"")).arg(mapPath);
+
+    }
+    return finalWorkingDir;
+}
+
+QString Command::executablePath() const
+{
+    QString finalcommand = finalCommand();
+
+    // removing parameters from command string
+    QString executableFilename(finalcommand.mid(0, finalcommand.lastIndexOf(QLatin1String(" ")) - 1));
```

Comment:
Of course handling quotes is a little bit more tricky, since you'd also need to handle `\"` within the executable path. Still of course a small function could be written for this.